src/types/state.rs

```diff
@@ -1,22 +1,30 @@
 use std::fmt;
 use std::io;
 use std::str;
+use std::str::FromStr;
+
+use serde::{Deserialize, Deserializer, Serialize};
+use serde::de::{Error, Unexpected};
 
 /// Possible battery state values.
 ///
 /// Unknown can mean either controller returned unknown,
 /// or not able to retrieve state due to some error.
-#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+#[derive(Debug, Eq, PartialEq, Copy, Clone, Serialize)]
+#[cfg_attr(feature = "config-schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "lowercase")]
 pub enum State {
     Unknown,
     Charging,
     Discharging,
     Empty,
     Full,
+}
 
-    // Awaiting for https://github.com/rust-lang/rust/issues/44109
-    #[doc(hidden)]
-    __Nonexhaustive,
+impl<'de> Deserialize<'de> for State {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer<'de> {
+        String::deserialize(deserializer).and_then(|s| State::from_str(&s).map_err(|_| D::Error::invalid_value(Unexpected::Str(&s), &"State")))
+    }
 }
```

Comment:
I've confirmed deriving works with https://github.com/starship/starship/pull/3941, you can use the derived implementation.
```suggestion
```